java-checks/src/main/java/org/sonar/java/checks/unused/UnusedPrivateMethodCheck.java

```diff
@@ -105,7 +111,6 @@ public void visitNode(Tree tree) {
         checkIfUnknown((MethodReferenceTree) tree);
         break;
       default:
-        throw new IllegalStateException("Unexpected subscribed tree.");
```

Comment:
I would add a comment here if you remove the exception, such as `// not registered - can not happen`